app.js

```diff
@@ -0,0 +1,168 @@
+
+    // check out data.info for metadata
+    if (destination === 'grid') {
+        characterGridData = parsedResponse.results;
+    } else {
+        favoriteCharactersData = Array.isArray(parsedResponse) ? parsedResponse : [parsedResponse];
+    }
+}
+
+// Looks at characterGridData and renders it in the grid.
+function showData(destination) {
+    if (destination === 'grid') {
+        cardContainer.innerHTML = '';
+    } else {
+        favoriteBar.innerHTML = '';
+    }
+    const collection = destination === 'grid' ? characterGridData : favoriteCharactersData;
+    console.log(collection);
+    collection.forEach((character) => {
+        if(destination === 'grid') {
+            renderCard(character);
+        } else {
+            renderFavoriteCard(character);
+        }
+    });
+}
+
+// Takes a single character's data and returns a card DOM element.
+function renderCard(characterData) {
+    let card = document.createElement('div');
+    card.id = characterData.id;
+    card.classList.add('characterCard');
+    card.addEventListener('click', favorite);
+    let cardTitle = document.createElement('div');
+    cardTitle.classList.add('characterNames');
+    let cardImg = document.createElement('img');
+    let favoriteIcon = document.createElement('img');
+    let cardText = document.createElement('ul');
+    let textSpecies = document.createElement('li');
+    let textStatus = document.createElement('li');
+    let textLocation = document.createElement('li');
+
+    favoriteIcon.src = checkFaves(card.id) ? 'img/favorite-heart.svg' : 'img/not-favorite-heart.svg'; 
+    favoriteIcon.classList.add('favoriteButton');
+    cardTitle.innerText = `${characterData.name}`;
+    cardImg.src = characterData.image;
+    textSpecies.innerText = `Species: ${characterData.species}`;
+    textStatus.innerText = `Status: ${characterData.status}`;
+    textLocation.innerText = `Location: ${characterData.location.name}`;
+    card.appendChild(cardImg);
+    card.appendChild(cardTitle);
+    card.appendChild(favoriteIcon);
+    card.appendChild(cardText);
+    cardText.appendChild(textSpecies);
+    cardText.appendChild(textStatus);
+    cardText.appendChild(textLocation);
+    cardContainer.appendChild(card);
+}
+
+function renderFavoriteCard(characterData) {
+    let favoriteCard = document.createElement('div');
+    favoriteCard.classList.add('favoriteCard');
+    favoriteCard.id = `favorite-card-${characterData.id}`;
+    let favoriteCardImg = document.createElement('img');
+    favoriteCardImg.src = characterData.image;
+    let favoriteCardName = document.createElement('div');
+    favoriteCardName.innerText = characterData.name;
+    let favoritePin = document.createElement('img');
+    favoritePin.src = 'img/thumb-tack.svg';
+    favoriteCard.appendChild(favoriteCardImg);
+    favoriteCard.appendChild(favoriteCardName);
+    favoriteCard.appendChild(favoritePin);
+    favoriteBar.appendChild(favoriteCard);
+}
```

Comment:
There is enough repetition between this function and the `renderCard` function that some of these things could be pulled out into their own small functions (or this could be combined with `renderCard`).

Reply:
Well taken - I think that breaking some of these DOM creation lines out into smaller, more easily generlized helper functons is a good middle ground compared to trying to make one main function for 'show data' and 'show faves' like we talked about on Sunday.